Mutex.php

```diff
@@ -0,0 +1,185 @@
+ *
+ * Redis Mutex requires redis version 2.6.12 or higher to work properly.
+ *
+ * It needs to be configured with a redis [[Connection]] that is also configured as an application component.
+ * By default it will use the `redis` application component.
+ *
+ * To use redis Mutex as the application component, configure the application as follows:
+ *
+ * ~~~
+ * [
+ *     'components' => [
+ *         'mutex' => [
+ *             'class' => 'yii\redis\Mutex',
+ *             'redis' => [
+ *                 'hostname' => 'localhost',
+ *                 'port' => 6379,
+ *                 'database' => 0,
+ *             ]
+ *         ],
+ *     ],
+ * ]
+ * ~~~
+ *
+ * Or if you have configured the redis [[Connection]] as an application component, the following is sufficient:
+ *
+ * ~~~
+ * [
+ *     'components' => [
+ *         'mutex' => [
+ *             'class' => 'yii\redis\Mutex',
+ *             // 'redis' => 'redis' // id of the connection application component
+ *         ],
+ *     ],
+ * ]
+ * ~~~
+ *
+ * @see \yii\mutex\Mutex
+ * @see http://redis.io/topics/distlock
+ *
+ * @author Sergey Makinen <sergey@makinen.ru>
+ * @since 2.0.6
+ */
+class Mutex extends \yii\mutex\Mutex
+{
+    /**
+     * @var int the number of seconds in which the lock will be auto released.
+     */
+    public $expire = 30;
+    /**
+     * @var string a string prefixed to every cache key so that it is unique. If not set,
+     * it will use a prefix generated from [[Application::id]]. You may set this property to be an empty string
+     * if you don't want to use key prefix. It is recommended that you explicitly set this property to some
+     * static value if the cached data needs to be shared among multiple applications.
+     */
+    public $keyPrefix;
+    /**
+     * @var Connection|string|array the Redis [[Connection]] object or the application component ID of the Redis [[Connection]].
+     * This can also be an array that is used to create a redis [[Connection]] instance in case you do not want do configure
+     * redis connection as an application component.
+     * After the Mutex object is created, if you want to change this property, you should only assign it
+     * with a Redis [[Connection]] object.
+     */
+    public $redis = 'redis';
+    /**
+     * @var array Redis lock values. Used to be safe that only a lock owner can release it.
+     */
+    private $_lockValues = [];
+
+    /**
+     * Initializes the redis Mutex component.
+     * This method will initialize the [[redis]] property to make sure it refers to a valid redis connection.
+     * @throws InvalidConfigException if [[redis]] is invalid.
+     */
+    public function init()
+    {
+        parent::init();
+        $this->redis = Instance::ensure($this->redis, Connection::className());
+        if ($this->keyPrefix === null) {
+            $this->keyPrefix = substr(md5(Yii::$app->id), 0, 5);
```

Comment:
I copied it to keep the implementation look like [other](https://github.com/yiisoft/yii2-redis/blob/master/Session.php#L94) Redis stuff. Let’s call this a convention.